src/external/electrs.ts

```diff
@@ -240,17 +238,19 @@ export class DefaultElectrsAPI implements ElectrsAPI {
      * @param amountAsBTC (Optional) Amount the recipient must receive
      * @returns Boolean value
      */
-    private utxoHasAtLeastAmount(utxo: UTXO | VOut, amount?: Big): boolean {
+    private utxoHasAtLeastAmount(utxo: UTXO | VOut, amount?: BTCAmount): boolean {
         if (amount) {
-            const expectedBtcAsSatoshi = btcToSat(amount);
-            if (utxo.value === undefined || expectedBtcAsSatoshi > new BN(utxo.value)) {
+            if (utxo.value === undefined) {
                 return false;
+            } else {
+                const utxoValue = BTCAmount.from.Satoshi(utxo.value);
+                return utxoValue >= amount;
```

Comment:
Are you sure this will work? I'd expect one would need to use `utxoValue.gte(amount)`

Reply:
I assumed that defining `gte`, `lte`, etc. members would override the default functionality of comparison operators. But this actually compares their string representations, which is wrong, thanks for pointing it out